### PR TITLE
[munit] Add new port

### PR DESCRIPTION
**Pull request by myd7349**

### CODE REVIEW

**Comment by myd7349** [comment]
https://github.com/microsoft/vcpkg/issues/7150

File: ports/munit/CONTROL [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Source: munit
+Version: 2019-05-11
```
Comment (LilyWangL) [inline ports/munit/CONTROL]: Can you please add the Homepage for this port in the CONTROL file?
```
Homepage: https://github.com/nemequ/munit
```

File: ports/munit/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,26 @@
+include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/munit/portfile.cmake]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/munit/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    OPTIONS
+        --backend=ninja
+)
+
+vcpkg_install_meson()
+
+vcpkg_copy_pdbs()
+
+# Handle copyright
+configure_file(${SOURCE_PATH}/COPYING ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)
```
Comment (LilyWangL) [inline ports/munit/portfile.cmake]: Can you please use command `file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Please fix build failures.

**Comment by LilyWangL** [comment]
Pinging @myd7349 for response. Is work still being done for this PR?

**Comment by LilyWangL** [comment]
Thanks for your PR. Please add [supports](https://github.com/microsoft/vcpkg/blob/e62d1361288e83eba786395b60361ab35ba83800/docs/maintainers/control-files.md#supports) to CONTROL. And please use [vcpkg_fail_port_install](https://github.com/microsoft/vcpkg/blob/5deea3b975fe62990973d73a102de818d83ba20c/docs/maintainers/vcpkg_fail_port_install.md) in portfile.cmake to description this port doesn't support some triplets.

**Comment by LilyWangL** [comment]
@myd7349 Thanks for your PR. Now this port build failed on triplet `x64_uwp` `arm64_windows` `arm_uwp` `x64_osx` `x64_linux`, if this port doesn't support these triplet. Can you use [Supports](https://github.com/microsoft/vcpkg/blob/e62d1361288e83eba786395b60361ab35ba83800/docs/maintainers/control-files.md#supports) and [vcpkg_fail_port_install](https://github.com/microsoft/vcpkg/blob/5deea3b975fe62990973d73a102de818d83ba20c/docs/maintainers/vcpkg_fail_port_install.md) to description this port doesn't support some triplets. And please modify _VCPKG_PATH/scripts/ci.baseline.txt_ to account for the build failures on unsupported triplets.

**Review by LilyWangL — APPROVED** [review]

**Comment by strega-nil** [comment]
We should probably add `vcpkg_find_acquire_program(Meson)` out of this.

File: ports/munit/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO nemequ/munit
+    REF da8f73412998e4f1adf1100dc187533a51af77fd
+    SHA512 cd08c1291a73487f15fdba7bf8675fea9177f0ec9766900f65efb5f00c662532a16499447e9087d304de34ff9138f47d04ebf18713f5aa8aacede22c5e23b98b
+    HEAD_REF master
+    PATCHES
+        fix-install.patch
+)
+
+vcpkg_configure_meson(
```
Comment (strega-nil) [inline ports/munit/portfile.cmake:15]: This should `vcpkg_find_acquire_program(MESON)`

**Review by strega-nil — CHANGES_REQUESTED** [review]
After adding the find_acquire_program, you should be able to run on x64-linux and x64-osx

File: ports/munit/portfile.cmake (hunk 4 above)
Comment (myd7349) [reply ports/munit/portfile.cmake:15]: Hi! @strega-nil Thanks for your review. But it seems that `vcpkg_configure_meson` has already done this internally: https://github.com/microsoft/vcpkg/blob/7e6e30ff8e9e0b823c212d4e61471989eebcbb0f/scripts/cmake/vcpkg_configure_meson.cmake#L94-L97

It seems that `vcpkg_find_acquire_program(MESON)` will do nothing but dump an error message if is not Windows.

**Comment by JackBoosY** [comment]
Does this port support Linux / OSX ?

**Comment by myd7349** [comment]
x64-linux:
```
-- Downloading https://github.com/nemequ/munit/archive/da8f73412998e4f1adf1100dc187533a51af77fd.tar.gz...
-- Extracting source /mnt/1/s/downloads/nemequ-munit-da8f73412998e4f1adf1100dc187533a51af77fd.tar.gz
-- Applying patch fix-install.patch
-- Using source at /mnt/1/s/buildtrees/munit/src/3a51af77fd-8670ab3989
CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:367 (message):
  Could not find meson.  Please install it via your package manager:

      sudo apt-get install meson
Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_meson.cmake:94 (vcpkg_find_acquire_program)
  ports/munit/portfile.cmake:15 (vcpkg_configure_meson)
  scripts/ports.cmake:90 (include)
```

x64-osx:
```
  File "/usr/local/bin/meson", line 2
    PYTHONPATH="/usr/local/Cellar/meson/0.54.1_2/lib/python3.8/site-packages" exec "/usr/local/Cellar/meson/0.54.1_2/libexec/bin/meson" "$@"
                                                                                 ^
SyntaxError: invalid syntax
```

I do not have a macbook at hand, so I can not figure out what happened to `x64-osx`. I am able to install `munit` on a virtual machine with CentOS which has `meson` pre-installed.

I can figure out a good solution currently, so I will disable them.

**Review by strega-nil — APPROVED** [review]

**Comment by strega-nil** [comment]
Alright, thanks @myd7349 :)

**Pull request merged**